include/utils/math.h

```diff
@@ -143,5 +143,15 @@ template<utils::floating_point T>
     return std::abs(value) < (std::numeric_limits<T>::epsilon() * 100.0);
 }
 
+/*!
+ * Calculates the sign of a numeric value, 1 if positive and -1 if negative
+ * @note 0 is also considered as positive
+ */
+template<utils::numeric T>
+[[nodiscard]] int8_t sign(T value)
+{
+    return value >= static_cast<T>(0) ? 1 : -1;
+}
```

Comment:
You could also use `copysign(1, <value>)` instead of calling this, but the lack of a proper `sign` in even modern C++ is (indeed) slightly strange.